contract/src/airdrop/prepare.js

```diff
@@ -0,0 +1,267 @@
+ * @property {bigint} basePayoutQuantity
+ * @property {{Keyword<Brand>}} brands
+ * @property {{Keyword<issuer>}} issuers
+ */
+
+/**
+ * @param {ZCF} zcf
+ * @param {{ purse: Purse, timer: import('@agoric/swingset-vat/tools/manual-timer.js').TimerService,}} privateArgs
+ * @param {import('@agoric/vat-data').Baggage} baggage
+ */
+export const start = async (zcf, privateArgs, baggage) => {
+  const zone = makeDurableZone(baggage, 'rootZone');
+
+  const terms = zcf.getTerms();
+
+  /** @type {ContractTerms} */
+  const {
+    AirdropUtils,
+    startTime,
+    basePayoutQuantity,
+    brands: { Token: tokenBrand },
+    issuers: { Token: tokenIssuer },
+  } = terms;
+
+  const createAmount = x => AmountMath.make(tokenBrand, x);
+
+  assert(startTime > 0n, 'startTime must be a BigInt larger than 0n.');
+  const claimedAccountsStore = zone.setStore('claimed users', {
+    durable: true,
+  });
+  const [{ stateMachine, states }, distributionSchedule, verify] =
+    await Promise.all([
+      E(AirdropUtils).getStateMachine(),
+      E(AirdropUtils).getSchedule(),
+      E(AirdropUtils).getVerificationFn(),
+    ]);
+
+  const cancelTokenMaker = makeCancelTokenMaker('airdrop-campaign');
+  await stateMachine.transitionTo(states.PREPARED);
+  console.log({ stateMachine, states });
+  const { purse: airdropPurse, timer } = privateArgs;
+  startupAssertion(airdropPurse, 'privateArgs.purse');
+  startupAssertion(timer, 'privateArgs.timer');
+
+  const makeUnderlyingAirdropKit = zone.exoClassKit(
+    'Airdrop Campaign',
+    {
+      helper: M.interface('Helper', {
+        makeAmountForClaimer: M.call().returns(M.any()),
+        cancelTimer: M.call().returns(M.promise()),
+        getDistributionEpochDetails: M.call().returns(M.record()),
+        updateDistributionMultiplier: M.call(M.any()).returns(M.promise()),
+        updateEpochDetails: M.call(M.any(), M.any()).returns(),
+      }),
+      creator: M.interface('Creator', {
+        createPayment: M.call().returns(M.any()),
+        prepareAirdropCampaign: M.call().returns(M.promise()),
+      }),
+      claimer: M.interface('Claimer', {
+        claim: M.call().returns(M.promise()),
+        view: M.call().returns(M.bigint()),
+        getStatus: M.call().returns(M.string()),
+      }),
+    },
+    /**
+     * @param {Purse} tokenPurse
+     * @param {Array} schedule
+     * @param stateMachine
+     * @param dsm
+     * @param store
+     */
+    (tokenPurse, schedule, dsm, store) => ({
+      currentCancelToken: null,
+      currentEpoch: 0,
+      distributionSchedule: schedule,
+      currentEpochEndTime: 0n,
+      basePayout: basePayoutQuantity,
+      earlyClaimBonus: schedule[0].tokenQuantity,
+      internalPurse: tokenPurse,
+      claimedAccounts: store,
+      dsm: Far('state machine', {
+        getStatus() {
+          return dsm.getStatus();
+        },
+        transitionTo(state) {
+          return dsm.transitionTo(state);
+        },
+      }),
+    }),
```

Comment:
`Far` objects don't fit in durable storage. I'm surprised this code runs. hm.

Reply:
it works because the `Far` object is not getting stored in the zone's baggage, and neither is any portion of this exo object. 

I modified `@agoric/zone/src/durable.js` to contain the following:
```js
// from zone/src/durable.js
export const makeDurableZone = baggage => {
  baggage || Fail`baggage required`;
  /** @type {import('.').Zone['exoClass']} */
  const exoClass = (...args) => prepareExoClass(baggage, ...args);
  /** @type {import('.').Zone['exoClassKit']} */
  const exoClassKit = (...args) => {
    console.group('---------- inside exoClassKit----------');
    console.log('------------------------');
    console.log('args::',args);
    console.log('------------------------');
    console.log(':: baggage.getSize()', baggage.getSize());
    console.log('------------------------');
    console.log(':: baggage.keys()', [...baggage.keys()]);
    console.log('------------------------');
    console.groupEnd();
    return prepareExoClassKit(baggage, ...args);
  };
  /** @type {import('.').Zone['exo']} */
  const exo = (...args) => prepareExo(baggage, ...args);

  const attachedStores = attachDurableStores(() => baggage);

  /** @type {import('.').Zone['subZone']} */
  const subZone = (label, options = {}) => {
    const subBaggage = provideDurableMapStore(baggage, label, options);
    return makeDurableZone(subBaggage);
  };

  return Far('durableZone', {
    exo,
    exoClass,
    exoClassKit,
    subZone,
    ...attachedStores,
  });
};
```

Executing the test code with these logs in place writes the following output to the console:

```sh
---------- inside exoClassKit----------
  ------------------------
  args:: [
    'Airdrop Campaign',
    {
      helper: {
        klass: 'Interface',
        interfaceName: 'Helper',
        methodGuards: [Object],
        sloppy: false
      },
      creator: {
        klass: 'Interface',
        interfaceName: 'Creator',
        methodGuards: [Object],
        sloppy: false
      },
      claimer: {
        klass: 'Interface',
        interfaceName: 'Claimer',
        methodGuards: [Object],
        sloppy: false
      }
    },
    [Function (anonymous)],
    {
      helper: {
        makeAmountForClaimer: [Function: makeAmountForClaimer],
        cancelTimer: [AsyncFunction: cancelTimer],
        getDistributionEpochDetails: [Function: getDistributionEpochDetails],
        updateEpochDetails: [Function: updateEpochDetails],
        updateDistributionMultiplier: [AsyncFunction: updateDistributionMultiplier]
      },
      creator: {
        createPayment: [Function: createPayment],
        prepareAirdropCampaign: [AsyncFunction: prepareAirdropCampaign]
      },
      claimer: {
        getStatus: [Function: getStatus],
        view: [Function: view],
        claim: [Function: claim]
      }
    }
  ]
  ------------------------
  :: baggage.getSize() 1
  ------------------------
  :: baggage.keys() [ 'claimed users' ]
  ------------------------
```

Reply:
I'm not sure if `durable: true` is needed above? 

Behavior aside - i'm of the belief that these operations don't need to be durably held. If the baggage contains a reference to both the current status string, as well as the allowedStateTransitions array, then that should be enough data to successfully reincarnate.

Reply:
keeping track of who has already claimed seems like it should be done durably.